xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -249,7 +249,9 @@ bool CAddonSettings::Load(const CXBMCTinyXML& doc)
       auto settingId = categoryId;
       if (!settingId.empty())
         settingId += ".";
-      settingId += setting->ValueStr();
+      auto id = setting->ToElement()->Attribute("id");
+      if (id)
+        settingId += id;
```

Comment:
don't think it's required, only check id so we don't crash on a nullptr. It should never return null unless xml is completely broken and we don't really handle that anywhere I think.